record_video.sh

```diff
@@ -37,33 +37,33 @@ do
 done
 ​
 ​
-# # ========================================================= 
+# ========================================================= 
 # # gstreamer command
-# # ========================================================= 
+# ========================================================= 
 ​
-# if [ "$HEIGHT" = 0 ]; then
-#     echo -e "No height specified. Exit.\n"
-#     usage
-#     exit 1
-# fi
+if [ "$HEIGHT" = 0 ]; then
+    echo -e "No height specified. Exit.\n"
+    usage
+    exit 1
+fi
 ​
-# if [ "$WIDTH" = 0 ]; then
-#     echo -e "No width specified. Exit.\n"
-#     usage
-#     exit 1
-# fi
+if [ "$WIDTH" = 0 ]; then
+    echo -e "No width specified. Exit.\n"
+    usage
+    exit 1
+fi
 ​
-# if [ "$FRAMERATE" = 0 ]; then
-#     echo -e "No frame rate specified. Exit.\n"
-#     usage
-#     exit 1
-# fi
+if [ "$FRAMERATE" = 0 ]; then
+    echo -e "No frame rate specified. Exit.\n"
+    usage
+    exit 1
+fi
 ​
-# if [ "$OUTPUTFILE" = 0 ]; then
-#     echo -e "No output file specified. Exit.\n"
-#     usage
-#     exit 1
-# fi
+if [ "$OUTPUTFILE" = 0 ]; then
+    echo -e "No output file specified. Exit.\n"
+    usage
+    exit 1
+fi
```

Comment:
@Shoki-Shoki Also, this statement exhaustively checks each one of the args - this is not very efficient, but in some cases is needed. In this case, we were looking for simplicity - so I think a better way would just be to check that we received 4 arguments.

Reply:
What do you think about the following? We can just check if the user has input 4 arguments, and print some help if not. To exhaustively check the args for the correct type could be very time consuming, not something we're looking for here as this script won't get used by many people, and those who will use it will be in the know. 
```suggestion
ARGUMENT_COUNT=$# 
if [[ "$ARGUMENT_COUNT -lt 4 ]]
then
    echo "Not enough args specified"
    usage
fi
```